src/components/Project/ProjectHeader/ProjectNameFields.tsx

```diff
@@ -1,23 +1,31 @@
 import { HookFormControlType } from '@/interfaces/formInterfaces';
 import { TextInput } from 'hds-react/components/TextInput';
 import { IconPenLine } from 'hds-react/icons';
-import { t } from 'i18next';
 import { FC, MouseEvent, useCallback, useState } from 'react';
 import { Control, Controller, FieldValues } from 'react-hook-form';
 import { IconButton } from '../../shared';
 import './styles.css';
+import { useAppSelector } from '@/hooks/common';
+import { selectMode } from '@/reducers/projectSlice';
+import { useTranslation } from 'react-i18next';
 
 interface IProjectNameFormProps {
   control: HookFormControlType;
 }
 
 const ProjectNameForm: FC<IProjectNameFormProps> = ({ control }) => {
+  const { t } = useTranslation();
   const [editing, setEditing] = useState(false);
-
-  const handleSetEditing = useCallback((e: MouseEvent<HTMLButtonElement>) => {
-    e.preventDefault();
-    setEditing((currentState) => !currentState);
-  }, []);
+  const projectMode = useAppSelector(selectMode);
+  const handleSetEditing = useCallback(
+    (e: MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      if (projectMode === 'edit') {
```

Comment:
instead of changing the click could we add the disabled prop to IconButton?

interface IIconButton {
...
disabled?: boolean;
}

const IconButton: FC = (props) => {
const { onClick, disabled, ...iconProps } = props;
return (
<button
onClick={onClick}
aria-label={iconProps.text || iconProps.label}
disabled={disabled || false}
>
...